src/main/kotlin/com/mdrobnak/lalrpop/psi/util/LpPsiUtil.kt

```diff
@@ -21,12 +21,12 @@ val List<LpSymbol>.selected: List<LpSymbol>
         this
     }
 
-fun List<LpSymbol>.computeType(context: LpTypeResolutionContext, arguments: LpMacroArguments): String {
-    val sel = selected
-    val joined = sel.joinToString(separator = ", ") { it.resolveType(context, arguments) }
-    return if (sel.size != 1) "($joined)"
-    else joined
-}
+fun List<LpSymbol>.computeType(context: LpTypeResolutionContext, arguments: LpMacroArguments): String =
+    selected.let { selected ->
+        val joined = selected.joinToString(separator = ", ") { it.resolveType(context, arguments) }
+        if (selected.size != 1) "($joined)"
+        else joined
+    }
```

Comment:
`selected` should be non-null already, so I don't think the `let` is necessary?

Reply:
I'd like to avoid doing the computation twice

Reply:
Ah, that's true. It's unfortunate that Kotlin hides the fact that this is a method call.